src/main/webapp/creatives/simid_protocol.js

```diff
@@ -251,27 +252,12 @@ export default class SimidProtocol {
       // to ever communicate with the player.
       console.log('Session creation was rejected.');
     }
-    this.generateSessionId_();
+
+    this.sessionId_ = generateSessionId();
```

Comment:
does it make sense to generate an ID for a session that got rejected -- perhaps only call generateSessionId() if the session is created?